test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,2361 @@
+          3. query information directly from the PBS database
+
+        :param hostname: the host to query
+        :type hostname: string or None
+        :param diag: the directory in which a pbs_diag is captured
+        :type diag: string or None
+        :returns: an instance of the class Server
+        """
+        server = Server(hostname, diag=diag)
+        if diag:
+            server.pbs_conf["PBS_HOME"] = diag
+            diag_time = self.get_diag_time(diag)
+            server.ctime = diag_time
+
+        if "PBS_SERVER" not in server.pbs_conf:
+            server.pbs_conf["PBS_SERVER"] = socket.gethostname()
+
+        if self.server_host is None:
+            self.server_host = server.hostname
+
+        return server
+
+    def __convert_flag_to_numeric(self, flag):
+        """
+        Convert a resource's flag attribute to its numeric equivalent
+
+        :param flag: the resource flag to convert
+        :type flag: string
+        :returns: numeric value of the resource flag
+        """
+        resc_flag = READ_WRITE
+        flag_ir = 0
+        if "q" in flag:
+            resc_flag |= ATR_DFLAG_RASSN
+        elif "f" in flag:
+            resc_flag |= ATR_DFLAG_FNASSN
+        elif "n" in flag:
+            resc_flag |= ATR_DFLAG_ANASSN
+        elif "h" in flag:
+            resc_flag |= ATR_DFLAG_CVTSLT
+        elif "r" in flag:
+            if flag_ir == 0:
+                resc_flag &= ~READ_WRITE
+                resc_flag |= NO_USER_SET
+            flag_ir += 1
+        elif "i" in flag:
+            resc_flag &= ~READ_WRITE
+            resc_flag |= (ATR_DFLAG_OPRD | ATR_DFLAG_OPWR |
+                          ATR_DFLAG_MGRD | ATR_DFLAG_MGWR)
+            flag_ir += 1
+        return resc_flag
+
+    def __convert_type_to_numeric(self, attr_type):
+        """
+        Convert a resource's type attribute to its numeric equivalent
+
+        :param attr_type: the type to convert
+        :type attr_type: string
+        :returns: Numeric equivalent of attr_type
+        """
+        return PBS_ATTR_TYPE_TO_INT[attr_type.strip()]
+
+    def __capture_trace_from_core(self, core_file_name, exec_path, out_path):
+        """
+        Capture stack strace from the core file specified
+
+        :param core_file_name: name of the core file
+        :type core_file_name: String
+        :param exec_path: path to the executable which generated the core
+        :type exec_path: String
+        :param out_path: ofile to print the trace out to
+        :type out_path: String
+        """
+        self.logger.debug("capturing stack trace from core file " +
+                          core_file_name)
+
+        # Create a gdb-python script to capture backtrace from core
+        gdb_python = "import gdb\n\
+        gdb.execute(\"core " + core_file_name + "\")\n\
+        o = gdb.execute(\"thread apply all bt\", to_string=True)\n\
+        print(o)\n\
+        gdb.execute(\"quit\")\n\
+        quit()"
```

Comment:
Good point.  Maybe have each string quoted?  Since "a" "b" "c" == "abc"  You still need the \'s but I find that less fragile.